tests/param/test_params.py

```diff
@@ -0,0 +1,408 @@
+from unittest.mock import Mock
+
+import bambi as bmb
+import pytest
+
+from hssm import HSSM, Link, Prior
+from hssm.config import Config
+from hssm.defaults import SupportedModels, default_model_config
+from hssm.param.param import Param
+from hssm.param.params import (
+    Params,
+    collect_user_params,
+    make_params,
+    make_param_from_user_param,
+    make_param_from_defaults,
+)
+from hssm.param.regression_param import RegressionParam
+from hssm.param.simple_param import DefaultParam, SimpleParam
+from hssm.param.user_param import UserParam
+
+
+def create_mock_model(
+    model_name: SupportedModels = "ddm",
+    loglik_kind="analytical",
+    global_formula=None,
+    link_settings=None,
+    prior_settings=None,
+):
+    def mock_config(model: SupportedModels, loglik_kind="analytical"):
+        if model not in default_model_config:
+            return lambda param: (None, None)
+
+        defaults = default_model_config[model]["likelihoods"][loglik_kind]
+
+        priors = defaults["default_priors"]
+        bounds = defaults["bounds"]
+
+        return lambda param: (priors.get(param), bounds.get(param))
+
+    config = Mock(
+        spec=Config,
+        get_defaults=mock_config(model_name, loglik_kind),
+    )
+
+    model = Mock(
+        model_name=model_name,
+        spec=HSSM,
+        list_params=default_model_config[model_name]["list_params"] + ["p_outlier"],
+        has_lapse=True,
+        loglik_kind=loglik_kind,
+        global_formula=global_formula,
+        link_settings=link_settings,
+        prior_settings=prior_settings,
+        model_config=config,
+        response_c="c(rt, response)",
+        additional_namespace={},
+    )
+    return model
+
+
+include = [
+    {"name": "v", "prior": {"name": "Normal", "mu": 0, "sigma": 1}},
+    {"name": "a", "prior": bmb.Prior("HalfNormal", sigma=1)},
+]
+
+
+def kw():
+    return {
+        "z": UserParam(prior=bmb.Prior("Normal", mu=0, sigma=1)),
+        "other_param": 1,
+    }
```

Comment:
Tried that. I haven't figured out a way for a new dictionary to be created every time the fixture is used by simply changing the scope of the fixture. Because the object is modified in-place, that caused some chaos. If there's a way for a new dictionary to be created every time a fixture is used, please let me know, and I'll definitely use a fixture.